DuckDuckGo/AIChat/AIChatMenuVisibilityConfigurable.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  AIChatMenuVisibilityConfigurable.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Combine
+
+protocol AIChatMenuVisibilityConfigurable {
+    var shouldDisplayApplicationMenuShortcut: Bool { get }
+    var shouldDisplayToolbarShortcut: Bool { get }
+    var shortcutURL: URL { get }
+    var valuesChangedPublisher: PassthroughSubject<Void, Never> { get }
+}
+
+final class AIChatMenuConfiguration: AIChatMenuVisibilityConfigurable {
```

Comment:
Yep, tests will be added once I add the feature to the app stack itself, right now I'm just adding small pieces of code to merge small PRs but the dependency stack is not 100% done since it needs to check privacy-config flags and whatnot